tests/scripts/test_determine_tests_to_run.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+
+from scripts.determine_tests_to_run import (
+    Import,
+    get_changed_source_files,
+    get_changed_test_files,
+    parse_imports,
+)
+
+
```

Comment:
Tested the simpler functions but left the more complex ones alone (for the time being).

The best way to test the dependency graph and related traversal would be to clone an older version of GE and perform the analysis on it. That feels out of scope for this stage but makes sense if we get closer to putting this in production.